backend/python/app/services/implementations/auth_service.py

```diff
@@ -173,37 +173,67 @@ def is_authorized_by_email(self, access_token, requested_email):
         except:
             return False
 
-    def is_translator(self, access_token, story_translation_content_id):
+    def is_translator(
+        self, access_token, story_translation_id=None, story_translation_content_id=None
+    ):
         return self._is_story_role(
-            access_token, story_translation_content_id, is_role_translator=True
+            access_token,
+            is_role_translator=True,
+            story_translation_id=story_translation_id,
+            story_translation_content_id=story_translation_content_id,
         )
 
-    def is_reviewer(self, access_token, story_translation_content_id):
+    def is_reviewer(
+        self, access_token, story_translation_id=None, story_translation_content_id=None
+    ):
         return self._is_story_role(
-            access_token, story_translation_content_id, is_role_translator=False
+            access_token,
+            is_role_translator=False,
+            story_translation_id=story_translation_id,
+            story_translation_content_id=story_translation_content_id,
         )
 
     def _is_story_role(
-        self, access_token, story_translation_content_id, is_role_translator
+        self,
+        access_token,
+        is_role_translator,
+        story_translation_id,
+        story_translation_content_id,
     ):
         try:
             decoded_id_token = firebase_admin.auth.verify_id_token(
                 access_token, check_revoked=True
             )
             user_id = self.user_service.get_user_id_by_auth_id(decoded_id_token["uid"])
+
             query_base = (
                 db.session.query(StoryTranslation.translator_id)
                 if is_role_translator
                 else db.session.query(StoryTranslation.reviewer_id)
             )
-            role_id = (
-                query_base.join(
-                    StoryTranslationContent,
-                    StoryTranslationContent.story_translation_id == StoryTranslation.id,
-                )
-                .filter(StoryTranslationContent.id == story_translation_content_id)
-                .first()
-            )[0]
+
+            if story_translation_id:
```

Comment:
nit: `role_id` is not declared prior to this if statement - this seems like it could be bug prone. Mind doing `role_id = None` before this if statement?